x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+      if (isOk(updateResult)) {
+        finalResults.push(updateResult.value);
+      } else {
+        const { id, type, error } = updateResult.error;
+
+        // this check is needed so error will be typed correctly for isConflictError
+        if (SavedObjectsErrorHelpers.isSavedObjectsClientError(error)) {
+          if (SavedObjectsErrorHelpers.isConflictError(error)) {
+            conflicts++;
+          } else {
+            logger.warn(
+              `Saved Object error updating task ${id}:${type} during claim: ${error.error}`,
+              logMeta
+            );
+            bulkErrors++;
+          }
+        } else {
+          logger.warn(`Error updating task ${id}:${type} during claim: ${error.message}`, logMeta);
+          bulkErrors++;
+        }
+      }
+    }
+  } catch (err) {
+    logger.warn(`Error updating tasks during claim: ${err}`, logMeta);
+  }
+
+  // separate update for removed tasks; shouldn't happen often, so unlikely
+  // a performance concern, and keeps the rest of the logic simpler
+  let removedCount = 0;
+  if (removedTasks.size > 0) {
+    const tasksToRemove = Array.from(removedTasks);
+    for (const task of tasksToRemove) {
+      task.status = TaskStatus.Unrecognized;
+    }
+
+    // don't worry too much about errors, we'll get them next time
+    try {
+      const removeResults = await taskStore.bulkUpdate(tasksToRemove, { validate: false });
+      for (const removeResult of removeResults) {
+        if (isOk(removeResult)) {
+          removedCount++;
+        } else {
+          const { id, type, error } = removeResult.error;
+          logger.warn(
+            `Error updating task ${id}:${type} to mark as unrecognized during claim: ${error.message}`,
+            logMeta
+          );
+        }
+      }
+    } catch (err) {
+      logger.warn(`Error updating tasks to mark as unrecognized during claim: ${err}`, logMeta);
+    }
+  }
+
+  const message = `task claimer claimed: ${finalResults.length}; stale: ${staleTasks.size}; conflicts: ${conflicts}; missing: ${missingTasks.size}; updateErrors: ${bulkErrors}; removed: ${removedCount};`;
+  logger.debug(message, logMeta);
+
+  const finalResult = {
+    stats: {
+      tasksUpdated: finalResults.length,
+      tasksConflicted: conflicts,
+      tasksClaimed: finalResults.length,
+    },
+    docs: finalResults,
+    timing: stopTaskTimer(),
+  };
+
+  for (const doc of finalResults) {
+    events$.next(asTaskClaimEvent(doc.id, asOk(doc), finalResult.timing));
+  }
+
+  return finalResult;
+}
+
+interface SearchAvailableTasksResponse {
+  docs: ConcreteTaskInstance[];
+  versionMap: Map<string, ConcreteTaskInstanceVersion>;
+}
+
+async function searchAvailableTasks({
+  definitions,
+  taskTypes,
+  removedTypes,
+  excludedTypes,
+  taskStore,
+  claimOwnershipUntil,
+  size,
+  taskMaxAttempts,
+}: OwnershipClaimingOpts): Promise<SearchAvailableTasksResponse> {
+  const searchedTypes = Array.from(taskTypes)
```

Comment:
nit: This list of types shouldn't change often right? Wondering if we could pre-compute this array and pass it in instead of computing every claim cycle.

Reply:
I don't think there's anywhere we can "pre-" compute it, but we could do it just once, the first time, then keep a copy.  I think this code is going to be changing when we fix the problem with marking  REMOVED_TYPES as `Unrecognized` in a better fashion (if we had a lot of these, they would starve candidate tasks).